hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -489,19 +489,24 @@ public OzoneManagerDoubleBuffer buildDoubleBufferForRatis() {
    */
   @Override
   public long takeSnapshot() throws IOException {
-    LOG.info("Current Snapshot Index {}", getLastAppliedTermIndex());
+    try {
+      awaitDoubleBufferFlush();
+    } catch (InterruptedException e) {
+      LOG.warn("double buffer wait is interrupted, ", e);
+      throw IOUtils.toInterruptedIOException("takeSnapshot is interrupted", e);
+    }
+    ozoneManager.getMetadataManager().getStore().flushDB();
+    TransactionInfo transactionInfo = ozoneManager.getMetadataManager()
+        .getTransactionInfoTable().get(TRANSACTION_INFO_KEY);
     TermIndex lastTermIndex = getLastAppliedTermIndex();
-    long lastAppliedIndex = lastTermIndex.getIndex();
+    LOG.info("Current Transaction Index {} lastAppliedTermIndex {}",
+        transactionInfo, lastTermIndex);
+    // update with lastAppliedTermIndex as snapshot have dependency with
+    // ratis commit index as triggered during notifyTermIndexUpdated after
+    // apply transaction. Can not use TRANSACTION_INFO_KEY from db
     snapshotInfo.updateTermIndex(lastTermIndex.getTerm(),
```

Comment:
Should it be updated with `transactionInfo` instead?  We should use the on disk state instead of the memory state for snapshots.

Reply:
> Should it be updated with `transactionInfo` instead? We should use the on disk state instead of the memory state for snapshots.

@szetszwo 
As discussed, I am updating index to double buffer using callback mechanism. Please have a look,
https://github.com/apache/ozone/pull/5857

We can discard this PR if above seems ok.

Reply:
Let's don't make the logic more complicated.  The direction is to remove both `applyTransactionMap` and `ratisTransactionMap` from the code.